### PR TITLE
Feature/48/create afgerond op function Percentage

### PR DESCRIPTION
**Pull request by NRBPerdijk**

### CODE REVIEW

File: engine/src/main/scala/org/scalarules/dsl/nl/grammar/AfrondingsWords.scala [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    def decimalen: DslEvaluation[Percentage] = {
+      DslEvaluation(factFilledCondition(afTeRondenPercentage), new Evaluation[Percentage] {
+        override def apply(c: Context): Option[Percentage] = {
+          Some(afTeRondenPercentage.toEval(c).get.afgerondOp(aantalDecimalen, roundingMode))
+        }
+      })
+    }
+  }
+
+  class Afgerond
```
Comment (jhkuperus) [inline engine/src/main/scala/org/scalarules/dsl/nl/grammar/AfrondingsWords.scala]: maybe make this `final` or `sealed`?

File: engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsTestBerekeningGlossary.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package org.scalarules.dsl.nl.grammar
+
+import org.scalarules.finance.nl.Percentage
+import org.scalarules.utils.Glossary
+
+object AfrondingsTestBerekeningGlossary extends Glossary {
+
+  val aantal = defineFact[Integer]()
```
Comment (jhkuperus) [inline engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsTestBerekeningGlossary.scala:8]: Note: the braces `()` here are optional. You can leave them out :) (if you want)

File: engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsTestBerekening.scala [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package org.scalarules.dsl.nl.grammar
+
+import org.scalarules.dsl.nl.grammar.AfrondingsWords._
+import org.scalarules.dsl.nl.grammar.AfrondingsTestBerekeningGlossary._
+
+import scala.language.postfixOps
+
+class AfrondingsTestBerekening extends Berekening (
+
+  Gegeven(afrondingsType is "halfNaarEven") Bereken
```
Comment (jhkuperus) [inline engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsTestBerekening.scala:10]: If you want to eliminate a `Fact`, you can actually have all these derivations use `Gegeven (altijd)`. Your tests only extract the useful value anyways.

Heck, you could even combine all your PBTs into just a single one ;)

File: engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsWordsTest.scala [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+class AfrondingsWordsTest extends PropSpec with Checkers {
+
+  property("halfNaarEven should match BigDecimal.RoundingMode.HALF_EVEN for positive doubles") {
+    check(
+      forAll (for {testGetal <- Gen.posNum[Double]} yield testGetal) { testGetal: Double =>
+
+        val context: Context = Map(startPercentage -> BigDecimal(testGetal).procent,
+                                    afrondingsType -> "halfNaarEven")
+
+        val uitkomst: Option[Percentage] = runAndExtractFact(context, new AfrondingsTestBerekening().berekeningen, afgerondPercentageHalfEven)
```
Comment (jhkuperus) [inline engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsWordsTest.scala:23]: If all derivations are run all the time, you can just `run` the engine here, and check every value returned in the context against its corresponding rounding mode.

After doing so, you might want to consider upping the number of checks performed on this property.

You could end with:

``` scala
all (
  afgerondPercentageHalfEven.toEval(context).isDefined,
  afgerondPercentageHalfEven.toEval(context).get == BigDecimal(testGetal).setScale(0, BigDecimal.RoundingMode.HALF_EVEN).procent :| "Rounding mode HALF_EVEN is broken",
  afgerondPercentageHalfUp.toEval(context).isDefined,
  ... // etc
)
```

File: engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsTestBerekeningGlossary.scala (hunk 2 above)
Comment (NRBPerdijk) [inline engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsTestBerekeningGlossary.scala:8]: I know, but leaving out the braces gives warnings and it's just a testglossary so I put them in anyway.

File: engine/src/main/scala/org/scalarules/dsl/nl/grammar/AfrondingsWords.scala (hunk 1 above)
Comment (NRBPerdijk) [inline engine/src/main/scala/org/scalarules/dsl/nl/grammar/AfrondingsWords.scala]: done

File: engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsTestBerekening.scala (hunk 3 above)
Comment (NRBPerdijk) [inline engine/src/test/scala/org/scalarules/dsl/nl/grammar/AfrondingsTestBerekening.scala:10]: As discussed: left it the way it was due to style preferences. :)

**Pull request merged**